### DISCUSSION TITLE
Unable to login as a user

### DISCUSSION THREAD

**Discussion by dasundev** [orchestral/testbench-dusk#99] [Q&A] [answered]
- Testbench Version: 8.24
- Laravel Version: 10.0
- PHP Version: 8.3
- Database Driver & Version: SQLite

### Description:
I tried to run a test as a user but it failed and captured this screenshot.

![checkout](https://github.com/orchestral/testbench-dusk/assets/54996800/3aae9699-ce86-4270-ac70-f59ab782f4d3)

### Steps To Reproduce:
```php
public function it_can_render_checkout_page()
{
    $user = User::factory()->create();

    $this->browse(function (Browser $browser) use ($user) {
        $browser->loginAs($user)
            ->visit('checkout')
            ->screenshot('checkout');
    });
}
```

**Comment by crynobone** [comment]
It's impossible to get this error on Laravel 10. 

https://laravel.com/docs/11.x/authentication#redirecting-unauthenticated-users

You can customise this via `workbench/bootstrap/app.php` but that's only going to be used for Laravel 11 and Testbench Dusk 9

**Reply by dasundev** [reply]
The `login` route is no longer encountering issues.

However, the following test case fails when executed.

```php
public function can_authenticate_user()
{
    $user = UserFactory::new()->create();

    $this->browse(static fn (Browser $browser) => $browser
        ->loginAs($user, 'web')
        ->assertAuthenticatedAs($user, 'web')
    );
}
```

Result:
```
PHPUnit 10.5.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.6
Configuration: /Users/dasun/Desktop/testbench-dusk-demo/phpunit.xml

F                                                                   1 / 1 (100%)

Time: 00:02.233, Memory: 34.50 MB

There was 1 failure:

1) Dasundev\TestbenchDuskDemo\Tests\Browser\AuthTest::can_authenticate_user
The currently authenticated user is not who was expected.
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
-Array &0 [
-    'id' => 1,
-    'className' => 'Illuminate\Foundation\Auth\User',
-]
+Array &0 []

/Users/dasun/Desktop/testbench-dusk-demo/vendor/laravel/dusk/src/Concerns/InteractsWithAuthentication.php:106
/Users/dasun/Desktop/testbench-dusk-demo/tests/Browser/AuthTest.php:24
/Users/dasun/Desktop/testbench-dusk-demo/vendor/laravel/dusk/src/Concerns/ProvidesBrowser.php:70
/Users/dasun/Desktop/testbench-dusk-demo/tests/Browser/AuthTest.php:22

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

**Reply by dasundev** [reply]
@crynobone I have created a demo repo, can you please check it?

https://github.com/dasundev/testbench-dusk-demo

**Reply by crynobone** [reply, marked as answer]
![CleanShot 2024-05-22 at 13 50 36](https://github.com/orchestral/testbench-dusk/assets/172966/d787d645-6798-4468-a196-ba16a9641777)

https://packages.tools/testbench-dusk/the-basic.html#supported-database

**Reply by dasundev** [reply]
I completely missed this. Yes, the `:memory:` database is not supported.

We need to use an `sqlite` database in `phpunit.xml`:
```xml
<env name="DB_CONNECTION" value="sqlite"/>
```
Thank you so much! You saved my day! ❤️
